server/mvcc/kvstore_compaction.go

```diff
@@ -90,7 +88,7 @@ func (s *store) scheduleCompaction(compactMainRev, prevCompactRev int64) (KeyVal
 		dbCompactionPauseMs.Observe(float64(time.Since(start) / time.Millisecond))
 
 		select {
-		case <-batchTicker.C:
+		case <-time.After(s.cfg.CompactionSleepInterval):
```

Comment:
- Curious why its defaulted to `10ms` today in the first place   ?

- As I understand it, compaction is taking longer than `10ms` in clusters with large data sets, leading to noticeable latency increases on client side (that's how this issue is surfaced iiuc). 
So, I was wondering if we can adjust `sleep interval dynamically`  based on `pending apply requests` instead of creating a room for `10ms` ?. The more apply requests pending, the longer the sleep interval, thus reducing client latency(for reads/writes) during high load scenarios and could be beneficial in large clusters. 
wdyt ?

Reply:
For q1, 10ms is introduced by https://github.com/etcd-io/etcd/pull/11034, and it seems like an empirical param https://github.com/etcd-io/etcd/pull/11021#issuecomment-520789359 (not the PR author's comment)

Reply:
For question 2, regardless of the duration of a compaction batch, the current implementation does not guarantee a full CompactionSleepInterval sleep. Given that CompactionSleepInterval is already configurable, I think we could  address this issue first and then introduce an adaptive sleep mode in a separate pull request :)

Reply:
For backport please stick to minimal change that will address the issue

Reply:
+1 to keep the minimal change in this PR.

For any improvement, let's address & discuss it separately.

Reply:
Yes, I am totally all for merging this for sure to fix the problem at hand. 

> let's address & discuss it separately.

I just left the community to understand what community thinks about this to see if it makes sense to go in that direction ?   I can probably cut a separate issue to channel this discussion.

Reply:
Sorry, do not get time to dig into your proposal yet. Please feel free to raise a separate ticket to present your thought/idea if you want. But you need to convince us that it's indeed a real issue and demo the real benefit to go in that direction. Overall I do not see a high priority to change it for now.